Ldelossa/srv6 monitor logging

ldelossa opened the pull request:
This PR renames "egress_policies.h" to "srv6.h" since only SRv6 related logic was left in this file.
Moving forward "srv6.h" will be the canonical file for SRv6 related datapath functionality. 

This PR also adds "srv6-encap" and "srv6-decap" monitor events, explaining to a debugger that SRv6 encap/decap successfully occurred and the subsequent packet is heading to the stack. 
This will cut down the time it takes to debug whether the SRv6 datapath processed an SRv6 VPN related packet. 

```release-note
Rename egress_policies.h to srv6.h and add SRv6 related trace reasons. 
```

ldelossa committed:
srv6,bpf: rename egress_policies.h to srv6.h

The only functions left in egress_policies.h are SRv6 related.
Let's rename this to 'srv6.h' and update references to the old file
name.

Signed-off-by: ldelossa <louis.delos@gmail.com>

ldelossa committed:
srv6,bpf: add srv6 related trace reasons

Include a trace reason for SRv6 encapsulation and decapsulation.

This greatly improves the debugging process, indicating whether SRv6
VPN related packets are processed by our datapath.

Signed-off-by: ldelossa <louis.delos@gmail.com>

YutaroHayakawa reviewed (changes requested):
Changes looks good to me, but in addition to this, can't we contain the information about the SID on the header we decapsulated? So that we can answer to the question like "is SID X correctly decapsulated?". Looks like send_trace_notify6 can take an "orig_addr". Can't we use it to store SID?

kaworu committed:
hubble: add support for TRACE_REASON_SRV6_{ENCAP,DECAP}

Consider encap/decap as egress/ingress (respectively) and both as
unknown reply ct status.

Signed-off-by: Alexandre Perrin <alex@isovalent.com>
